### PR TITLE
DEV: use db in export

### PR DESCRIPTION
**Pull request by licode**
in case we want to use other functions in db.

### CODE REVIEW

**Comment by licode** [comment]
api change doc added. @tacaswell

File: doc/source/api_change.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
+.. _api_changes:
+
+=============
+ API changes
+=============
+
+
+v0.5.2
```
Comment (tacaswell) [inline doc/source/api_change.rst]: Should be 0.6

File: suitcase/tests/test_hdf5.py [hunk 2]
```diff
@@ -24,7 +25,7 @@ def shallow_header_verify(hdf_path, header, mds, fields=None, stream_name=None,
             if stream_name is not None:
                 if stream_name != descriptor.name:
                     continue
-            table_load = mds.get_events_table(descriptor)
+            table_load = db.mds.get_events_table(descriptor)
```
Comment (tacaswell) [inline suitcase/tests/test_hdf5.py]: change to `db.get_table` please.

File: suitcase/tests/test_nexus.py [hunk 3]
```diff
@@ -25,7 +25,7 @@ def shallow_header_verify(hdf_path, header, mds, fields=None, stream_name=None,
             if stream_name is not None:
                 if stream_name != descriptor.name:
                     continue
-            table_load = mds.get_events_table(descriptor)
+            table_load = db.mds.get_events_table(descriptor)
```
Comment (tacaswell) [inline suitcase/tests/test_nexus.py]: `db.get_table`

**Comment by licode** [comment]
This should be ready to go. @tacaswell ?

**Comment by tacaswell** [comment]
Where does the LiveSpec code live?  Will that need to be updated?

**Comment by licode** [comment]
spec file looks good. I changed mds to db, and no others need to be updated.

File: suitcase/spec.py [hunk 4]
```diff
@@ -523,27 +521,27 @@ def specscan_to_document_stream(scan, mds, validate=False, check_in_broker=False
         metadatastore.  You will need to call find_* yourself to determine
         if it does exist
     """
-    if mds is None and check_in_broker:
+    if db.mds is None and check_in_broker:
```
Comment (tacaswell) [inline suitcase/spec.py]: if `db is None` this will raise an attribute error.

File: suitcase/spec.py [hunk 5]
```diff
@@ -592,7 +590,7 @@ def _get_timestamp(dt):
 }
 
 
-def _check_and_update_document(doc_name, doc_dict, mds):
+def _check_and_update_document(doc_name, doc_dict, db=None):
```
Comment (tacaswell) [inline suitcase/spec.py]: This one can not be optional

File: suitcase/tests/test_hdf5.py [hunk 6]
```diff
@@ -7,7 +7,8 @@
 import pytest
 
 
-def shallow_header_verify(hdf_path, header, mds, fields=None, stream_name=None, use_uid=True):
+def shallow_header_verify(hdf_path, header, fields=None,
+                          stream_name=None, use_uid=True, db=None):
```
Comment (tacaswell) [inline suitcase/tests/test_hdf5.py]: db can not be optional here either.

File: suitcase/hdf5.py [hunk 7]
```diff
@@ -41,22 +42,30 @@ def export(headers, filename, mds,
...
         Otherwise group name is created based on beamline id and run id.
+    db : databroker object, optional
+        db should be included in hdr.
     """
     if isinstance(headers, Header):
         headers = [headers]
+
+    try:
+        db = headers[0].db
+    except KeyError:
```
Comment (tacaswell) [inline suitcase/hdf5.py]:
```python
try:
   db = headers[0].db
except KeyError, AttributeError:
   pass
if db is None:
    raise RuntimeError(...)
```

**Pull request merged**